packages/block-library/src/terms-query/edit/inspector-controls/index.js

```diff
@@ -46,24 +46,26 @@ export default function TermsQueryInspectorControls( {
 		( _taxonomy ) => _taxonomy.slug === taxonomy
 	)?.hierarchical;
 
-	const isTaxonomyMatchingTemplate = templateSlug?.startsWith(
-		// `Tags` are a special case in WP template hierarchy.
-		taxonomy === 'post_tag' ? 'tag' : taxonomy
-	);
-
-	// Only display the inherit control if the taxonomy is hierarchical and matches the current template.
+	// Display the inherit control when we're in a taxonomy-related
+	// template (category, tag, or custom taxonomy).
 	const displayInheritControl =
-		isTaxonomyHierarchical && isTaxonomyMatchingTemplate;
+		templateSlug?.startsWith( 'taxonomy' ) ||
+		templateSlug?.startsWith( 'category' ) ||
+		templateSlug?.startsWith( 'tag' ) ||
```

Comment:
Could we more strictly match the template slugs: `tag` and `tag-xxx` etc.? Because strictly speaking `tagline` or something like that would be a custom template.

Reply:
We could try to be super strict by exactly fetching taxonomies and terms, but I think it's fine for now because the toggle displays informative messages and its default value is **not** to inherit.

For me this can be handled in a follow up if needed, but I believe it's going to be enough as is.

Reply:
Why not just add `templateSlug === 'tag' || templateSlug?.startsWith( 'tag-' )` etc. separately? That's good enough, we don't need to fetch the taxonomies, it fits exactly what the hierarchy does. If we don't do this it will lead to bugs.

Reply:
Sounds good. I'll update.

Reply:
Startswith has to check `tag-` though, and same for the other two?

Reply:
How about this: https://github.com/WordPress/gutenberg/pull/72313/commits/520bf55f9ca2b93ace730abb9fa60abd4384b6d3?

Reply:
Thanks!